daemon/cmd/kube_proxy_replacement.go

```diff
@@ -339,6 +339,10 @@ func initKubeProxyReplacementOptions() (strict bool) {
 
 // handleNativeDevices tries to detect bpf_host devices (if needed).
 func handleNativeDevices(strict bool) {
+	if option.Config.EnableWireguard {
+		option.Config.DirectRoutingDevice = wireguardTypes.IfaceName
```

Comment:
If we happen to be overriding user-specified values here, we should make that visible in the logs.